### DISCUSSION TITLE
Pager with only Next/Previous button (version 2)

### DISCUSSION THREAD

**Discussion by minhhungit** [serenity-is/Serenity#6350] [Tricks and Samples]
Hi, 
Base on issue https://github.com/volkanceylan/Serenity/issues/4458 
I created a wiki page here for pager version 2 https://github.com/serenity-is/Serenity/wiki/Custom-pager-with-only-next-previous-mode-(version-2)

Code of version 2 is more cleaner than version 1 and easier for maintaining 

I'll be happy to hear any ideas to improve this feature, thank you 

(This post also let people who used first version know about the update to apply to their project if necessary 😃 )

**Comment by kilroyFR** [comment]
Very nice thanks. Does this work with Serenity 4.x ? Looks like not because of use of IRow.

I am getting errors on IRow (not being found) and missing implementation of `public bool ActivateFor()`

```
  public class UseNextPreviousPagerBehavior : IImplicitBehavior, IListBehavior
  {
    IRow _row;

    public bool ActivateFor(IRow row)
```

Thanks again

**Reply by minhhungit** [reply]
that code is for .net core version, for .net framework, use this:

```csharp
Row _row;

public bool ActivateFor(Row row)
{
    var mt = row as IUseNextPreviousPager;
    if (mt == null)
        return false;

    _row = row;
    return true;
}
```

**Reply by kilroyFR** [reply]
Thanks again !
